misk/src/main/kotlin/misk/security/authz/AccessInterceptor.kt

```diff
@@ -0,0 +1,56 @@
+package misk.security.authz
+
+import misk.Action
+import misk.ApplicationInterceptor
+import misk.Chain
+import misk.MiskCaller
+import misk.exceptions.UnauthenticatedException
+import misk.exceptions.UnauthorizedException
+import misk.scope.ActionScoped
+import javax.inject.Inject
+import kotlin.reflect.full.findAnnotation
+
+class AccessInterceptor private constructor(
+  private val allowedServices: Set<String>,
+  private val allowedRoles: Set<String>,
+  private val caller: ActionScoped<MiskCaller?>
+) : ApplicationInterceptor {
+
+  override fun intercept(chain: Chain): Any {
+    val caller = caller.get() ?: throw UnauthenticatedException()
```

Comment:
nice! does it make sense to include an error message like `no caller found` or is giving away too much info?